frontend/src/ToDoList.tsx

```diff
@@ -1,29 +1,42 @@
 import React, {useEffect, useState} from "react";
-import {ToDoItem, NewItem} from "./itemModel";
-import ToDoGalleryItem from "./ToDoGalleryItem";
+import {ToDoItem} from "./itemModel";
+import ToDoGalleryItem from "./GalleryItem/ToDoGalleryItem";
 import './ToDoList.css';
+import { useTranslation } from 'react-i18next';
+import NewItem from "./newItem/NewItem";
 
 export default function ToDoList() {
     const [searchTerm, setSearchTerm] = useState('');
     const [toDoList, setToDoList] = useState([] as Array<ToDoItem>);
-    const [newItem, setNewItem] = useState({title: 'test', description: '', done: false} as NewItem);
-
-    const [titleField, setTitleField] = useState('');
-    const [descriptionField, setDescriptionField] = useState('');
+    const [errorMessage, setErrorMessage] = useState('');
 
+    const { t } = useTranslation();
+    
+    useEffect(() => getAll(), [])
```

Comment:
Der Build schlägt an dieser Stelle fehl, weil `getAll` das `t` verwendet und `t` variabel ist. `t` muss also in deps Array und das `getAll` muss in ein `useCallback`. Es würde dann ungefähr so aussehen. Außerdem würde ich das `setSearchTerm` in einem `then` Block ausführen, so dass es erst ausgeführt wird, wenn die Response wieder beim client ist.
```javascript
const getAll = useCallback(() => {
    fetch(`${process.env.REACT_APP_BASE_URL}/todoitems/getall`)
        .then(response => response.json())
        .then((list: Array<ToDoItem>) => setToDoList(list))
        .catch(() => console.log('oopsie - getAll'))
        .then(response => {
            if (response.status >= 200 && response.status < 300) {
                return response.json();
            }
            throw new Error(`${t('get-all-error')}, ${t('error')}: ${response.status}`)
        })
        .then((list: Array<ToDoItem>) => {
            setToDoList(list);
            setErrorMessage('');
            setSearchTerm(''):
        })
        .catch(e => {console.log(e.message); setErrorMessage(e.message)})
}, [t]);

useEffect(() => getAll(), [getAll])
```